### PR TITLE
Ballmer peak!

### PR DESCRIPTION
**Pull request by VirtualSatai**

### CODE REVIEW

**Comment by DelusionalLogic** [comment]
That's a strange travis problem!

**Comment by VirtualSatai** [comment]
Yep, [I'll try something](http://s2.quickmeme.com/img/4a/4a102f0529cd0acddfa88a8dea431001d8e671f9def8ecaa6992bc455cee56f4.jpg)

File: stregsystem/booze.py [hunk 1]
```diff
@@ -66,16 +66,31 @@ def alcohol_bac_timeline(gender, weight, now, alcohol_timeline):
         current += alcohol_bac_increase(gender, weight, ml)
 
         # last_time can never be None after the first iteration
-        assert(last_time is not None)
+        assert (last_time is not None)
 
     # Since we return if the list is empty we must have some last time
-    assert(last_time is not None)
+    assert (last_time is not None)
```
Comment (DelusionalLogic) [inline stregsystem/booze.py]: Alright, but lets remove the parens too then.

File: stregsystem/booze.py [hunk 2]
```diff
@@ -66,16 +66,31 @@ def alcohol_bac_timeline(gender, weight, now, alcohol_timeline):
...
     return current
+
+
+# Ballmer peak: 1.337 +/- 0.05
+BALLMER_PEAK_MEAN = 1.337
+BALLMER_PEAK_LOWER_LIMIT = BALLMER_PEAK_MEAN - 0.05
+BALLMER_PEAK_UPPER_LIMIT = BALLMER_PEAK_MEAN + 0.05
+
+
+def time_to_ballmer_peak_expiry(bac):
```
Comment (DelusionalLogic) [inline stregsystem/booze.py]: I'd say this should be generalized to some other method taking a current and future bac, and returning the timedelta to reach that future bac.

File: stregsystem/booze.py [hunk 3]
```diff
@@ -66,16 +66,31 @@ def alcohol_bac_timeline(gender, weight, now, alcohol_timeline):
...
     if current < 0:
         current = 0
 
     return current
+
+
+# Ballmer peak: 1.337 +/- 0.05
+BALLMER_PEAK_MEAN = 1.337
+BALLMER_PEAK_LOWER_LIMIT = BALLMER_PEAK_MEAN - 0.05
+BALLMER_PEAK_UPPER_LIMIT = BALLMER_PEAK_MEAN + 0.05
```
Comment (DelusionalLogic) [inline stregsystem/booze.py]: I like the globals, but let's only use them in this file.

File: stregsystem/views.py [hunk 4]
```diff
@@ -105,6 +106,11 @@ def quicksale(request, room, member, bought_ids):
         return render(request, 'stregsystem/error_stregforbud.html', locals())
 
     promille = member.calculate_alcohol_promille()
+    is_ballmer_peaking = BALLMER_PEAK_LOWER_LIMIT < promille < BALLMER_PEAK_UPPER_LIMIT
```
Comment (DelusionalLogic) [inline stregsystem/views.py]: Maybe extract most of this (at least the is_ballmer_peaking) into a function in booze.

File: stregsystem/booze.py (hunk 2 above)
Comment (VirtualSatai) [reply stregsystem/booze.py]: I did consider that, but didn't want to bother with doing regression on previous data or something of that kind.

File: stregsystem/views.py (hunk 4 above)
Comment (VirtualSatai) [reply stregsystem/views.py]: Sure, why not

File: stregsystem/booze.py (hunk 2 above)
Comment (DelusionalLogic) [reply stregsystem/booze.py]: When would you need that? Wouldn't it just be exactly the same as you have now?

**Pull request merged**